lib/nimble_strftime.ex

```diff
@@ -312,8 +312,9 @@ defmodule NimbleStrftime do
   end
 
   # Microseconds
-  defp format_modifiers("f" <> rest, width, pad, datetime, format_options, acc) do
-    result = datetime.microsecond |> elem(0) |> Integer.to_string() |> pad_leading(width, pad)
+  defp format_modifiers("f" <> rest, _width, _pad, datetime, format_options, acc) do
+    {microsecond, precision} = datetime.microsecond
+    result = microsecond |> Integer.to_string() |> String.pad_leading(6, "0") |> binary_part(0, max(precision,1))
```

Comment:
```suggestion
    result = microsecond |> Integer.to_string() |> String.pad_leading(6, "0") |> binary_part(0, max(precision, 1))
```

Otherwise the formatter will fail later. :) Speaking about it... we should probably setup CI!